tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -854,16 +855,122 @@ impl<'a, F> fmt::Debug for FieldFnVisitor<'a, F> {
     }
 }
 
+// === printing synthetic Span events ===
+
+/// Select the degree to which tracing spans are logged as events
+#[derive(Debug)]
+pub enum FmtSpan {
+    /// spans are ignored (this is the default)
+    None,
+    /// one event per enter/exit of a span
+    Active,
+    /// one event when the span is dropped
+    Close,
+    /// events at all points (new, enter, exit, drop)
+    Full,
+}
```

Comment:
Is there a way to declare this to be open, so that we can add more variants without needing to bump the major version?

Reply:
We should use the same technique used for `Level` — where the enum is a private type that's a field of a public struct, and each variant is defined as an associated constant:
https://github.com/tokio-rs/tracing/blob/8628f8906ee57db5448047ac013030d96e674835/tracing-core/src/metadata.rs#L241-L262

This way, the variants can be matched, but not exhaustively.

We could also use `#[non_exhaustive]` but this is more portable to older Rust versions.

Reply:
Yep, looks good: done!